WooCommerce/Classes/POS/ViewModels/CartViewModel.swift

```diff
@@ -22,14 +22,26 @@ final class CartViewModel: CartViewModelProtocol {
         return itemsInCart.isEmpty
     }
 
-    init() {
+    private var analytics: PointOfSaleAnalytics?
+
+    init(analytics: PointOfSaleAnalytics? = nil) {
+        self.analytics = analytics
+
         cartSubmissionPublisher = cartSubmissionSubject.eraseToAnyPublisher()
         addMoreToCartActionPublisher = addMoreToCartActionSubject.eraseToAnyPublisher()
     }
 
     func addItemToCart(_ item: POSItem) {
         let cartItem = CartItem(id: UUID(), item: item, quantity: 1)
         itemsInCart.append(cartItem)
+
+        guard let analytics = analytics else {
+            return
+        }
+        // TODO:
+        // - Update analytics property to not be Optional
+        // - Move events to a pos-wide enum
+        analytics.track("test_item_added_to_cart", properties: [:], error: nil)
```

Comment:
I prefixed this with `test_`, but AFAU we can already use the _final_ event names. Thoughts?

Reply:
Yes, I think so too.